src/main/java/io/vertx/core/file/impl/FileResolver.java

```diff
@@ -375,11 +379,16 @@ private ClassLoader getClassLoader() {
     return cl;
   }
 
-  private synchronized void setupCacheDir(String id) {
-    String cacheDirName = fileCacheDir + "/file-cache-" + id;
-    cacheDir = new File(cacheDirName);
-    if (!cacheDir.mkdirs()) {
-      throw new IllegalStateException("Failed to create cache dir: " + cacheDirName);
+  private synchronized void setupCacheDir() {
+    FileAttribute<Set<PosixFilePermission>> ownerPermissions =
+        PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------"));
+
+    try {
+      Path cacheDirBase = new File(cacheDirBaseName).toPath();
+      Files.createDirectories(cacheDirBase, ownerPermissions);
```

Comment:
I think we shouldn't try to create the `cacheDirBase` but only verify that it exists. Otherwise we'd fall in the same trap described in #3510

Reply:
There are 3 use cases:
A) `vertx.cacheDirBase` is not set, then `java.io.tmpdir` is used, this is an existing directory, `Files.createDirectories` is a no-op, this works as before.
B) The sysop sets `vertx.cacheDirBase` to an existing directory. `Files.createDirectories` is a no-op, this works as before. It was and will be the sysop's task to set directory permissions if this is to be used by multiple users.
C) The sysop sets `vertx.cacheDirBase` to a non-existing directory. If we don't want to introduce a breaking change vertx should create that directory. This has been working and will work if only a single user starts vertx applications. There are many use cases where this is a reasonable approach. (Restricting directory access to owner as suggested below makes this a secure solution.)
I don't see any advantages not to create the directory; what benefit do we get from this breaking change?